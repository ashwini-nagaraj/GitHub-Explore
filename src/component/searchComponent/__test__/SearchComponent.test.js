import React from "react";
import ReactDOM from "react-dom";
import Index from "../index";
import { render , cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashig", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Index></Index>, div);
});

it("renders loading correctly", () => {
  const { getByTestId } = render(
    <Index>
    </Index>
  );
  const homepage = getByTestId("homepage");
  const img = getByTestId("img");
  expect(homepage).toContainElement(img);
});

it("matches snapshot",()=>{
    const tree=  renderer.create(<Index></Index>).toJSON();
     expect(tree).toMatchSnapshot();
 })