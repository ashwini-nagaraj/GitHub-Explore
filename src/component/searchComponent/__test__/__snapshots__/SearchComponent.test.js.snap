// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="homepage-wrapper"
  data-testid="homepage"
>
  <img
    alt="#"
    className="logo-image"
    data-testid="img"
    src="https://github.githubassets.com/images/modules/logos_page/Octocat.png"
  />
  <div
    className="search-input-box"
    data-testid="search"
  >
    <input
      className="search-txt"
      data-testid="search-box"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter Github Username + ENTER!"
      type="text"
    />
    <button
      className="search-btn"
      data-testid="search-button"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-search"
      />
    </button>
  </div>
</div>
`;
